event_groups.c

```diff
@@ -625,43 +625,49 @@ EventBits_t xEventGroupSetBits( EventGroupHandle_t xEventGroup,
 
 void vEventGroupDelete( EventGroupHandle_t xEventGroup )
 {
-    EventGroup_t * pxEventBits = xEventGroup;
-    const List_t * pxTasksWaitingForBits = &( pxEventBits->xTasksWaitingForBits );
+    EventGroup_t * pxEventBits;
+    const List_t * pxTasksWaitingForBits;
 
-    vTaskSuspendAll();
+    if ( NULL != xEventGroup )
     {
-        traceEVENT_GROUP_DELETE( xEventGroup );

```

Comment:
I thiink it is actually important that the trace hook is the first line in the function here?